module/VuFind/src/VuFind/Db/Entity/SearchEntityInterface.php

```diff
@@ -40,4 +42,170 @@
  */
 interface SearchEntityInterface extends EntityInterface
 {
+    /**
+     * Get identifier (returns null for an uninitialized or non-persisted object).
+     *
+     * @return ?int
+     */
+    public function getId(): ?int;
+
+    /**
+     * Get user.
+     *
+     * @return int
+     */
+    public function getUserId(): int;
+
+    /**
+     * Set user.
+     *
+     * @param int $user User
+     *
+     * @return SearchEntityInterface
+     */
+    public function setUserId(int $user): SearchEntityInterface;
+
+    /**
+     * Get session.
+     *
+     * @return ?string
+     */
+    public function getSessionId(): ?string;
+
+    /**
+     * Set session.
+     *
+     * @param ?string $session_id Session id
+     *
+     * @return SearchEntityInterface
+     */
+    public function setSessionId(?string $session_id): SearchEntityInterface;
+
+    /**
+     * Get created.
+     *
+     * @return DateTime
+     */
+    public function getCreated(): DateTime;
+
+    /**
+     * Set created date.
+     *
+     * @param DateTime $dateTime Created date
+     *
+     * @return SearchEntityInterface
+     */
+    public function setCreated(DateTime $dateTime): SearchEntityInterface;
+
+    /**
+     * Get title.
+     *
+     * @return ?string
+     */
+    public function getTitle(): ?string;
+
+    /**
+     * Set title.
+     *
+     * @param ?string $title Title
+     *
+     * @return SearchEntityInterface
+     */
+    public function setTitle(?string $title): SearchEntityInterface;
+
+    /**
+     * Get saved.
+     *
+     * @return int
+     */
+    public function getSaved(): int;
+
+    /**
+     * Set saved.
+     *
+     * @param int $saved Saved
+     *
+     * @return SearchEntityInterface
+     */
+    public function setSaved(int $saved): SearchEntityInterface;
+
+    /**
+     * Get the search object from the row.
+     *
+     * @return \VuFind\Search\Minified
+     */
+    public function getSearchObject(): \VuFind\Search\Minified;
+
+    /**
+     * Set search object.
+     *
+     * @param \VuFind\Search\Minified $search_object Search object
+     *
+     * @return SearchEntityInterface
+     */
+    public function setSearchObject(\VuFind\Search\Minified $search_object): SearchEntityInterface;
+
+    /**
+     * Get checksum.
+     *
+     * @return ?int
+     */
+    public function getChecksum(): ?int;
+
+    /**
+     * Set checksum.
+     *
+     * @param ?int $checksum Checksum
+     *
+     * @return SearchEntityInterface
+     */
+    public function setChecksum(?int $checksum): SearchEntityInterface;
+
+    /**
+     * Get notification frequency.
+     *
+     * @return int
+     */
+    public function getNotificationFrequency(): int;
+
+    /**
+     * Set notification frequency.
+     *
+     * @param int $notification_frequency Notification frequency
```

Comment:
We generally use $camelCase instead of $snake_case for PHP variables. It's probably worth adjusting all of the parameters to use $camelCase style for consistency.

I realize that this looks a bit inconsistent in combination with the snake_case class properties used by the Laminas\Db rows (which we are forced to use because of how its database abstraction works). Fortunately, this inconsistency will go away when we finish the Doctrine migration.